EIPS/eip-4610.md

```diff
@@ -0,0 +1,98 @@
+---
+eip: 4610
+title: ERC-721 Delegator Extension
+description: Enables ERC-721 to be loaned
+author: AFKDAO (@AFKDAO)
+discussions-to: https://ethereum-magicians.org/t/delegator-extension-that-enables-erc721-to-be-loaned/7885
+status: Draft
+type: Standards Track
+category: ERC
+created: 2021-12-28
+requires: 721
+---
+
+## Abstract
+The optional ERC-721 Delegator Extension provides a standardized function which could enable ERC-721 to be loaned. 
+
+## Motivation
+The ERC-721 standard does not realize the separation of ownership and use rights, which makes it difficult to realize the loan of ERC-721. Solving this problem will help make ERC-721 more widely used.
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+**Every ERC-4610 compliant contract must implement the `ERC721` interfaces** (subject to "caveats" below):
+
+```solidity
+pragma solidity ^0.8.0;
+
+import "./IERC721.sol";
+
+interface IERC4610 is IERC721 {
+
+    /**
+     * @dev Emitted when `owner` or `approved account` enables `setDelegator` to manage the `tokenId` token.
+     */
+    event SetDelegator(address indexed caller, address indexed delegator, uint256 indexed tokenId);
+
+    /**
+     * @dev Set or remove `delegator` for the owner.
+     * The delegator has no direct permission, just an additional attribute.
+     *
+     * Requirements:
+     *
+     * - The `delegator` cannot be the caller.
+     * - `tokenId` must exist.
+     *
+     * Emits an {SetDelegator} event.
+     */
+    function setDelegator(address delegator, uint256 tokenId) external;
+
+    /**
+     * @dev Returns the delegator of the `tokenId` token.
+     *
+     * Requirements:
+     *
+     * - `tokenId` must exist.
+     */
+    function delegatorOf(uint256 tokenId) external view returns (address);
+
+    /**
+     * @dev Safely transfers `tokenId` token from `from` to `to`
+     * `delegator` won't be cleared if `reserved` is true.
+     *
+     * Requirements:
+     *
+     * - `from` cannot be the zero address.
+     * - `to` cannot be the zero address.
+     * - `tokenId` token must exist and be owned by `from`.
+     * - If the caller is not `from`, it must be approved to move this token by either {approve} or {setApprovalForAll}.
+     * - If `to` refers to a smart contract, it must implement {IERC721Receiver-onERC721Received}, which is called upon a safe transfer.
+     * - If `reserved` is true, it won't clear the `delegator`.
+     *
+     * Emits a {Transfer} event.
+     */
+    function safeTransferFrom(
+        address from,
+        address to,
+        uint256 tokenId,
+        bool reserved
+    ) external;
+
+}
```

Comment:
While you outline a new `safeTransferFrom` function here which has a bool for determining if a delegator is carried over from a transfer, this would not do away with the `transfer` or `safeTransfer` functions in the ERC-721 specification. (In other words, since the arguments are different, it wouldn't override those functions, it wouls just overload `safeTransferFrom` another time, as it's already overloaded in the 721 spec.)

What should happen when a user transfers using `transferFrom` or the original `safeTransferFrom`s?

Reply:
> 

I will also add the intent of this overloaded function in the EIP